src/App.tsx

```diff
@@ -3,14 +3,25 @@ import './App.css';
 import { useState } from 'react';
 import { UserData } from './Interfaces';
 import LoginForm from './Login';
-import MenuBar from './components/layout/MenuBar';
+// import { getUser } from './apiCalls';
+import MenuBar from './components/layout/MenuBar'
+// import { BrowserRouter } from 'react-router-dom';
+import { useUserLoggedContext } from './context/UserLoggedContext.tsx';
 import UserInformation from './components/pages/userInformation';
 import { Route, Routes, Navigate } from 'react-router-dom';
 import Companies from './components/companies/Companies';
 import NewCompany from './components/companies/NewCompany';
 
+// interface UserInfo {
+//   id: number,
+//   username: string,
+//   email: string,
+//   token: number,
+//   role: string[]
+// }
+
 function App() {
-  const [userId, setUserId] = useState<number | null>(null);
+  const { isLoggedIn, clearUserLogged } = useUserLoggedContext()
```

Comment:
clearUserLogged shouldn't be in main. The purpose behind context is to prevent prop drilling. Not to say that props shouldn't exist, but if we can prevent prop drilling then we should. This is probably something that should be extracted out of main. Currently it does make sense because we don't have a homescreen, but either way it would probably be best to have a log out button on the menuBar rather than the homescreen, and have that function reside as a part of menubar